executor/linux_fork_server_executor.cpp

```diff
@@ -0,0 +1,585 @@
+                );
+        }
+	else
+            dest.resize( cur_size + read_stat );
+        return read_stat != 0;
+    }
+}
+
+/**
+ * Precondition:
+ *  - input_fd has a file descriptor that is set by LinuxForkServerExecutor::SetupIO()
+ * Postcondition:
+ *  - (1) The contents of file refered by input_fd matches to fuzz ( the contents of buf with length of len )
+ *  - (2) To execute process that satisfies following requirements ( process that satisfies them is notated as "competent process" ).
+ *      - Providing command and commandline arguments specified by constructor argument argv.
+ *      - Environment variables satisfy following condition:
+ *        * If fuzzuf's basic block coverage is used, receive environment variable __FUZZUF_SHM_ID, then open shared memory which has the specified id.
+ *        * If AFL's edge coverage is used, receive environment variable __AFL_SHM_ID, then open shared memory which has the specified id.
+ *      - TODO: Specify the requirements of environemnt variables (Items on instrument tool. Currently, only major things are mentioned, as so that can increase for implementing new features. )
+ *      - If Executor::stdin_mode is true, use input_fd as a standard input.
+ *    (3) Competent process stops execution after the time specified by timeout_ms. As the exception, if the value is 0, the time limit is determined using member variable exec_timelimit_ms.
+ *    (4) When the method exits, it is triggered by self or third-party signals.
+ *    (5) Assign process ID of competent process to child_pid
+ *      - This postcondition is needed for third-party to exit competent process
+ *        Is there need for third-party to exit the process in future? -> It might be used in the case signal handler is in use.
+ *        The postcondition is left for future extension.
+ */
+
+void LinuxForkServerExecutor::Run(const u8 *buf, u32 len, u32 timeout_ms) {
+    // locked until std::shared_ptr<u8> lock is used in other places
+    while (IsFeedbackLocked()) {
+        usleep(100);
+    }
+
+    // if timeout_ms is 0, then we use exec_timelimit_ms;
+    if (timeout_ms == 0) timeout_ms = exec_timelimit_ms;
+
+    // Aliases
+    ResetSharedMemories();
+    if (record_stdout_and_err) {
+        stdout_buffer.clear();
+        stderr_buffer.clear();
+    }
+
+    WriteTestInputToFile(buf, len);
+
+    //#if 0
+    // TODO: Since the information priority is Trace level that is less important than Debug, it should be hidden when runlevel is Debug.
+    DEBUG("Run: ");
+    DEBUG("%s ", cargv[0]);
+    std::for_each( cargv.begin(), cargv.end(), []( const char* v ) { DEBUG("%s ", v); } );
+    DEBUG("\n")
+    //#endif
+
+    // This structure is shared by both parent process and child process.
+    // Since execv never returns on success, it is initialized as success(0), then set value on failed.
+    auto child_state = fuzzuf::utils::interprocess::create_shared_object(
+      fuzzuf::executor::ChildState{ 0, 0 }
+    );
```

Comment:
This is not needed

Reply:
Thank you for suggestion. I removed these lines.